### ISSUE TITLE
Duplicates and "False Positives"

### ISSUE THREAD

**Issue by Green-Sky** [mull-project/mull#782]
Hi, I tried this awesome tool on https://github.com/skypjack/entt and had some troubles.
The tool reported
1. a lot of Duplicates and
2. most reported Mutants, when replicated by hand, did not survive the test suite (the same executable).

The aforementioned Project uses `GTest`. I speculate that `mull` tests each test-case separately?

Steps to replicate:
- use ubuntu 18.04 and install `mull` via provided repo: https://mull.readthedocs.io/en/latest/Installation.html#install-on-ubuntu
- install `clang-9` (might want to test different clang versions)
- build `entt` with `CC=clang-9 CXX=clang++-9 cmake .. -DENTT_BUILD_TESTING=ON  -DCMAKE_CXX_FLAGS="-fembed-bitcode" -DCMAKE_EXPORT_COMPILE_COMMANDS=ON` in `./build`
- setup mull junk removal etc (https://mull.readthedocs.io/en/latest/fmtlibTutorial.html#junk-mutations)

my `mull` command than looks like this:
```
mull-cxx \
	--test-framework=GoogleTest \
	-compdb-path compile_commands.json \
	-compilation-flags="\
	-isystem /usr/bin/../lib64/gcc/x86_64-linux-gnu/8/../../../../include/c++/8 \
	-isystem /usr/bin/../lib64/gcc/x86_64-linux-gnu/8/../../../../include/x86_64-linux-gnu/c++/8 \
	-isystem /usr/bin/../lib64/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/backward \
	-isystem /usr/local/include \
	-isystem /usr/lib/llvm-9/lib/clang/9.0.0/include \
	-isystem /usr/include/x86_64-linux-gnu \
	-isystem /usr/include" \
	-ld-search-path=/lib/x86_64-linux-gnu \
	-ld-search-path=/usr/lib/x86_64-linux-gnu \
	--exclude-path _deps \
	-mutators cxx_all \
	test/view
```

<details>
  <summary>example output for `test/view`</summary>
  

```
[info] Extracting bitcode from executable (threads: 1)
       [################################] 1/1. Finished in 27ms
[info] Loading bitcode files (threads: 4)
       [################################] 4/4. Finished in 182ms
[info] Compiling instrumented code (threads: 4)
       [################################] 4/4. Finished in 10ms
[info] Loading dynamic libraries (threads: 1)
       [################################] 1/1. Finished in 0ms
[info] Searching tests (threads: 1)
       [################################] 1/1. Finished in 2ms
[info] Preparing original test run (threads: 1)
       [################################] 1/1. Finished in 906ms
[info] Running original tests (threads: 4)
       [################################] 28/28. Finished in 74ms
[info] Applying function filter: no debug info (threads: 4)
       [################################] 2395/2395. Finished in 11ms
[info] Applying function filter: file path (threads: 4)
       [################################] 2395/2395. Finished in 10ms
[info] Instruction selection (threads: 4)
       [################################] 2344/2344. Finished in 16ms
[info] Searching mutants across functions (threads: 4)
       [################################] 2344/2344. Finished in 39ms
[info] Applying filter: no debug info (threads: 4)
       [################################] 3010/3010. Finished in 10ms
[info] Applying filter: file path (threads: 4)
       [################################] 3010/3010. Finished in 11ms
[info] Applying filter: junk (threads: 4)
       [################################] 3010/3010. Finished in 1530ms
[info] Prepare mutations (threads: 1)
       [################################] 1/1. Finished in 0ms
[info] Cloning functions for mutation (threads: 4)
       [################################] 4/4. Finished in 1089ms
[info] Removing original functions (threads: 4)
       [################################] 4/4. Finished in 483ms
[info] Redirect mutated functions (threads: 4)
       [################################] 4/4. Finished in 11ms
[info] Applying mutations (threads: 1)
       [################################] 179/179. Finished in 10ms
[info] Compiling original code (threads: 4)
       [################################] 4/4. Finished in 10ms
[info] Running mutants (threads: 4)
       [################################] 179/179. Finished in 10556ms
[info] Survived mutants (69/179):
/home/green/workspace/github/entt/src/entt/entity/registry.hpp:131:25: warning: Survived: Replaced < with <= [cxx_lt_to_le]
        return (!(index < pools.size()) || !pools[index].pool) ? nullptr : static_cast<const storage_type<Component> *>(pools[index].pool.get());
                        ^
/home/green/workspace/github/entt/src/entt/entity/registry.hpp:370:21: warning: Survived: Replaced < with <= [cxx_lt_to_le]
        return (pos < entities.size() && entities[pos] == entity);
                    ^
/home/green/workspace/github/entt/src/entt/entity/storage.hpp:173:24: warning: Survived: Replaced 'a = b' with 'a = 42' [cxx_assign_const]
        instances[pos] = std::move(other);
                       ^
/home/green/workspace/github/entt/src/entt/entity/storage.hpp:172:14: warning: Survived: Replaced 'T a = b' with 'T a = 42' [cxx_init_const]
        auto other = std::move(instances.back());
             ^
/home/green/workspace/github/entt/src/entt/entity/sparse_set.hpp:153:45: warning: Survived: Replaced & with | [cxx_and_to_or]
        return size_type{(to_integral(entt) & traits_type::entity_mask) / entt_per_page};
                                            ^
/home/green/workspace/github/entt/src/entt/entity/sparse_set.hpp:153:73: warning: Survived: Replaced / with * [cxx_div_to_mul]
        return size_type{(to_integral(entt) & traits_type::entity_mask) / entt_per_page};
                                                                        ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:894:45: warning: Survived: Replaced --x with ++x [cxx_pre_dec_to_pre_inc]
                for(auto pos = size(); pos; --pos) {
                                            ^
/home/green/workspace/github/entt/src/entt/entity/registry.hpp:131:25: warning: Survived: Replaced < with <= [cxx_lt_to_le]
        return (!(index < pools.size()) || !pools[index].pool) ? nullptr : static_cast<const storage_type<Component> *>(pools[index].pool.get());
                        ^
/home/green/workspace/github/entt/src/entt/entity/registry.hpp:388:49: warning: Survived: Replaced >> with << [cxx_rshift_to_lshift]
        return version_type(to_integral(entity) >> traits_type::entity_shift);
                                                ^
/home/green/workspace/github/entt/src/entt/entity/registry.hpp:149:47: warning: Survived: Replaced & with | [cxx_and_to_or]
        const auto entt = to_integral(entity) & traits_type::entity_mask;
                                              ^
/home/green/workspace/github/entt/src/entt/entity/storage.hpp:173:24: warning: Survived: Replaced 'a = b' with 'a = 42' [cxx_assign_const]
        instances[pos] = std::move(other);
                       ^
/home/green/workspace/github/entt/src/entt/entity/storage.hpp:172:14: warning: Survived: Replaced 'T a = b' with 'T a = 42' [cxx_init_const]
        auto other = std::move(instances.back());
             ^
/home/green/workspace/github/entt/src/entt/entity/storage.hpp:154:45: warning: Survived: Replaced - with + [cxx_sub_to_add]
            const auto pos = size_type(index-1u);
                                            ^
/home/green/workspace/github/entt/src/entt/entity/registry.hpp:144:61: warning: Survived: Replaced & with | [cxx_and_to_or]
        available = entity_type{to_integral(entities[curr]) & traits_type::entity_mask};
                                                            ^
/home/green/workspace/github/entt/src/entt/entity/registry.hpp:144:19: warning: Survived: Replaced 'a = b' with 'a = 42' [cxx_assign_const]
        available = entity_type{to_integral(entities[curr]) & traits_type::entity_mask};
                  ^
/home/green/workspace/github/entt/test/entt/entity/view.cpp:315:15: warning: Survived: Replaced 'a = b' with 'a = 42' [cxx_assign_const]
        check = false;
              ^
/home/green/workspace/github/entt/test/entt/entity/view.cpp:329:15: warning: Survived: Replaced 'a = b' with 'a = 42' [cxx_assign_const]
        check = false;
              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with >= [cxx_lt_to_ge]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with <= [cxx_lt_to_le]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with <= [cxx_lt_to_le]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/sparse_set.hpp:513:23: warning: Survived: Replaced 'T a = b' with 'T a = 42' [cxx_init_const]
        for(size_type pos{}; pos < count; ++pos) {
                      ^
/home/green/workspace/github/entt/src/entt/entity/sparse_set.hpp:513:34: warning: Survived: Replaced < with >= [cxx_lt_to_ge]
        for(size_type pos{}; pos < count; ++pos) {
                                 ^
/home/green/workspace/github/entt/src/entt/entity/sparse_set.hpp:513:43: warning: Survived: Replaced ++x with --x [cxx_pre_inc_to_pre_dec]
        for(size_type pos{}; pos < count; ++pos) {
                                          ^
/home/green/workspace/github/entt/src/entt/entity/sparse_set.hpp:513:23: warning: Survived: Replaced 'T a = b' with 'T a = 42' [cxx_init_const]
        for(size_type pos{}; pos < count; ++pos) {
                      ^
/home/green/workspace/github/entt/src/entt/entity/sparse_set.hpp:513:34: warning: Survived: Replaced < with >= [cxx_lt_to_ge]
        for(size_type pos{}; pos < count; ++pos) {
                                 ^
/home/green/workspace/github/entt/src/entt/entity/sparse_set.hpp:513:43: warning: Survived: Replaced ++x with --x [cxx_pre_inc_to_pre_dec]
        for(size_type pos{}; pos < count; ++pos) {
                                          ^
/home/green/workspace/github/entt/test/entt/entity/view.cpp:581:20: warning: Survived: Replaced < with <= [cxx_lt_to_le]
        return lhs < rhs;
                   ^
/home/green/workspace/github/entt/test/entt/entity/view.cpp:589:20: warning: Survived: Replaced > with >= [cxx_gt_to_ge]
        return lhs > rhs;
                   ^
/home/green/workspace/github/entt/test/entt/entity/view.cpp:589:20: warning: Survived: Replaced > with <= [cxx_gt_to_le]
        return lhs > rhs;
                   ^
/home/green/workspace/github/entt/test/entt/entity/view.cpp:599:20: warning: Survived: Replaced < with <= [cxx_lt_to_le]
        return lhs < rhs;
                   ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with >= [cxx_lt_to_ge]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with <= [cxx_lt_to_le]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with >= [cxx_lt_to_ge]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with <= [cxx_lt_to_le]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:231:21: warning: Survived: Replaced 'T a = b' with 'T a = 42' [cxx_init_const]
        std::size_t pos{};
                    ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/storage.hpp:85:20: warning: Survived: Replaced --x with ++x [cxx_pre_dec_to_pre_inc]
            return --index, *this;
                   ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:233:159: warning: Survived: Replaced x++ with x-- [cxx_post_inc_to_post_dec]
        (static_cast<void>(std::get<storage_type<Component> *>(pools) == cpool ? nullptr : (other[pos] = std::get<storage_type<Component> *>(pools), other[pos++])), ...);
                                                                                                                                                              ^
/home/green/workspace/github/entt/src/entt/entity/storage.hpp:154:45: warning: Survived: Replaced - with + [cxx_sub_to_add]
            const auto pos = size_type(index-1u);
                                            ^
/home/green/workspace/github/entt/test/entt/entity/view.cpp:783:15: warning: Survived: Replaced 'a = b' with 'a = 42' [cxx_assign_const]
        check = false;
              ^
/home/green/workspace/github/entt/test/entt/entity/view.cpp:817:15: warning: Survived: Replaced 'a = b' with 'a = 42' [cxx_assign_const]
        check = false;
              ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with >= [cxx_lt_to_ge]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with <= [cxx_lt_to_le]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with >= [cxx_lt_to_ge]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with <= [cxx_lt_to_le]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with >= [cxx_lt_to_ge]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with <= [cxx_lt_to_le]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with >= [cxx_lt_to_ge]
            return lhs->size() < rhs->size();
                               ^
/home/green/workspace/github/entt/src/entt/entity/view.hpp:226:32: warning: Survived: Replaced < with <= [cxx_lt_to_le]
            return lhs->size() < rhs->size();
                               ^
[info] Mutation score: 61%
[info] Total execution time: 15006ms
```
</details>

**Comment by AlexDenisov** [comment]
Hi @Green-Sky, thanks for the report.
This is an unfortunate issue that is caused by an (arguable) bug in our reporting mechanism.
This happens in cases when the code resides in headers, especially with C++ templates.
Here is an explanation:

Let's say you have the following code in a header:
```
template <typename T>
T doSomething(T arg) {
  return arg + arg;
}
```

Then, you use this code with two different types which lead to the instantiation of the template with two types:

```
ASSERT_EQ(doSomething(42.0), 84.0);
ASSERT_EQ(doSomething(42), 84);
```

The resulting machine code gets two versions of `doSomething`, e.g.:

```llvm
define float @_doSomething_float(float %arg) {
  %r = fadd float %arg, %arg
  ret float %r
}
``` 

and

```llvm
define i32 @_doSomething_int(i32 %arg) {
  %r = add i32 %arg, %arg
  ret i32 %r
}
``` 

As Mull works on the lower level it sees these two functions and generates two distinct mutants, one for each function.
It may happen that the test is killed for type `float`, but not killed for type `int`. Mull, therefore. reports two mutants, one of which is killed, even though both of the mutants correspond to the same source location in the original program.

While this explanation doesn't justify the presence of the bug it is at least explains its nature.
The easiest way to fix it is on the level of each reporter, but then we need to modify each reporter.
The right way to fix it is on the level of Mull so that it merges both mutants before passing them to a reporter. This solution, however, is less trivial since it requires more changes in the main Mull's logic.

This issue will be fixed, though I cannot give any ETA.

**Comment by stanislaw** [comment]
Simple example that reproduces this:

```cpp
#include <gtest/gtest.h>

#include <iostream>

template <typename T>
T doSomething(T arg1, T arg2) {
  return arg1 + arg2;
}

TEST(Sandbox, Test) {
  ASSERT_EQ(doSomething(2, 3), 5);
  ASSERT_EQ(doSomething(2.0, 0.0), 2.0);
}
```

One mutant is killed (2 + 3 == 5), one is not (2.0 + 0.0 == 2.0).

**Comment by AlexDenisov** [comment]
Hi @Green-Sky, just a follow up: this issue will be addressed in the next release.

**Comment by AlexDenisov** [comment]
Hi @Green-Sky, the fix has just landed. You can already give it a try if you wish:

Change the apt sources to `deb https://dl.bintray.com/mull-project/ubuntu-18 nightly main` and then install the corresponding version:

```
sudo apt-get update
sudo apt-get install mull=0.9.0-pr829
```
You'd also need to change the mull invocation to the following:

```
mull-cxx \
	-linker=clang++-9 \
	-compdb-path compile_commands.json \
	-compilation-flags="\
	-isystem /usr/bin/../lib64/gcc/x86_64-linux-gnu/8/../../../../include/c++/8 \
	-isystem /usr/bin/../lib64/gcc/x86_64-linux-gnu/8/../../../../include/x86_64-linux-gnu/c++/8 \
	-isystem /usr/bin/../lib64/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/backward \
	-isystem /usr/local/include \
	-isystem /usr/lib/llvm-9/lib/clang/9.0.0/include \
	-isystem /usr/include/x86_64-linux-gnu \
	-isystem /usr/include" \
	-exclude-path _deps \
	-mutators cxx_all \
	test/view
```

(Note the absence of `-ld-search-path` and `-test-framework`, and the new `-linker` option).

Please open a new issue if you hit any other problem.

**Comment by Green-Sky** [comment]
:O wow you guys work fast...

> (Note the absence of -ld-search-path and -test-framework, and the new -linker option).

So, I already tried the new version yesterday :+1: , so I got that part. I also had to add `--linker-flags="-pthread"`, because gtest :shrug:.

And it now works :heart:!!
Thanks for your time and work you guys are putting into this project. This has a lot of potential after all.

now the only duplicates are from having multiple test binaries :smirk: ...
